lua/entities/gmod_wire_expression2/core/custom/ftrace.lua

```diff
@@ -253,72 +288,196 @@ local function trcWorld(oFTrc, eE, vP, vA)
 	local trS, trE = oFTrc.mTrI.start, oFTrc.mTrI.endpos
 	trS:Set(eP); trE:Set(eA:Forward())
 	trE:Mul(oFTrc.mLen); trE:Add(trS)
-	-- http://wiki.garrysmod.com/page/util/TraceLine
+	-- https://wiki.facepunch.com/gmod/util.TraceLine
 	util.TraceLine(oFTrc.mTrI); return oFTrc
 end
 
-local function dumpItem(oFTrc, sNam, sPos)
+local function updateEarSize(oFTrc)
+	if(not oFTrc) then return nil end
+	local tE, iE = oFTrc.mFlt.Ear, 1
+	while(tE[iE]) do local vE = tE[iE]
+		if(isValid(vE)) then iE = iE + 1
+		else table.remove(tE, iE) end
+	end; oFTrc.mFlt.Size = (iE - 1)
+	return oFTrc
+end
+
+--[[
+ * Moves only the entities from source to destination
+ * oFTrc > Reference to tracer object
+ * tData > Source data table to read from ( entity array )
+ * bTab  > When enabled process as table instead of array
+ * bID   > When enabled the source table contains entity ID
+]]
+local function putFilterEar(oFTrc, tData, bTab, bID)
+	if(not oFTrc) then return nil end
+	local tE = oFTrc.mFlt.Ear
+	if(bTab) then local iD = 1
+		while(tData[iD]) do
+			local vE, iE, eE = tData[iD]
+			if(bID) then
+				iE = math.floor(tonumber(vE) or 0)
+				if(iE > 0) then eE = Entity(iE) end
+			else eE = vE end
+			if(isValid(eE, "Entity")) then
+				table.insert(tE, eE)
+			end; iD = iD + 1
+		end
+	else
+		for iD, vE in pairs(tData) do local iE, eE
+			if(bID) then
+				iE = math.floor(tonumber(vE) or 0)
+				if(iE > 0) then eE = Entity(iE) end
+			else eE = vE end
+			if(isValid(eE, "Entity")) then
+				table.insert(tE, eE)
+			end
+		end
+	end; return updateEarSize(oFTrc)
+end
+
+--[[
+ * Moves only the entities from source to destination
+ * oFTrc > Reference to tracer object
+ * tData > Source data table to read from ( function filter )
+]]
+local function putFilterFnc(oFTrc, tData)
+	if(not oFTrc) then return nil end
+	if(not tData) then return oFTrc end
+	for iD = 1, tData.Size do local vD = tData[iD]
+		-- Transfer the SKIP configuration for the table
+		if(vD.SKIP) then for key, val in pairs(vD.SKIP) do
+			setFncFilter(oFTrc, vD.CALL, "SKIP", key, true) end end
+		-- Transfer the ONLY configuration for the table
+		if(vD.ONLY) then for key, val in pairs(vD.ONLY) do
+			setFncFilter(oFTrc, vD.CALL, "ONLY", key, true) end end
+	end
+	-- Transsfer referencies from the entity array
+	for key, val in pairs(tData.Ent.SKIP) do
+		oFTrc.mFnc.Ent.SKIP[key] = val end
+	for key, val in pairs(tData.Ent.ONLY) do
+		oFTrc.mFnc.Ent.ONLY[key] = val end
+	return oFTrc -- Return the reference of the tracer
+end
+
+--[[
+ * Returns copy array for the entity filter list
+ * oFTrc > Reference to tracer object
+ * bID   > When enabled the table contains entity ID
+]]
+local function getEntityList(oFTrc, bID)
+	if(not oFTrc) then return nil end
+	local tO, iO = {}, 0
+	local tE, iD = oFTrc.mFlt.Ear, 1
+	while(tE[iD]) do local vE = tE[iD]

```

Comment:
Would prefer a for loop instead of while

Reply:
H, more or less like the first option. I just do not care how long the array is ( skip using `#` to count them all ) so I loop element by element..

Reply:
`for iD, vE in ipairs(tE) do`?

Reply:
`while(tE[iD]) do local vE = tE[iD]`